docs/library/datadisplay/spinner.md

```diff
@@ -0,0 +1,49 @@
+---
+components:
+    - rx.spinner
+    - rx.button
+---
+
+# Spinner
+
+Spinner is used to display an animated loading indicator when a task is in progress.
+
+```python exec
+import reflex as rx
+```
+
+```python demo
+rx.spinner()
+```
+
+## Basic Examples
+
+```python demo
+rx.vstack(
+    rx.hstack(
+        rx.spinner(size="1"),
+        rx.spinner(size="2"),
+        rx.spinner(size="3"),
+        align="center",
+        gap="1em"
+    )
+)
+```
+
+## Demo with buttons
+
+Buttons have their own loading prop that automatically composes a spinner.
+
+```python demo
+rx.button(
+        rx.spinner(
+            rx.icon("bookmark"),
+            loading=True
```

Comment:
The loading prop doesn't do anything here. Its only for the button which internally displays the spinner